File: client/Network/WebSocketParams.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Simplerpgkataclient.Network
{
    public enum MoveDirection
    {
        None = 0,
        Up = 1,
        Right = 2,
        Down = 3,
        Left = 4
    }

    public enum ActionType
    {
        None = 0,
        Movement = 1,
        Attack = 2
    }

    public class WebSocketParams
    {
        public string sessionId { get; set; }
        public int characterId { get; set; }
        public float positionX { get; set; } 
        public float positionY { get; set; }
        public int moveDirection { get; set; }
        public double hp { get; set; }
        public double initHp { get; set; }
        public int level { get; set; }
        public int classId { get; set; }
        public bool isConnected { get; set; }
        public int targetId { get; set; }
        public int actionType { get; set; }
    }
}

File: server/src/main/java/my/plaground/Domain/DTO/WebSocketParams.java
package my.plaground.Domain.DTO;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import my.plaground.Domain.ActionType;
import my.plaground.Domain.CharacterClass;
import my.plaground.Domain.MoveDirection;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class WebSocketParams {

    private String sessionId;

    private int characterId;
    private double positionX;
    private double positionY;
    @JsonProperty("moveDirection")
    @JsonFormat(shape = JsonFormat.Shape.NUMBER_INT)
    private MoveDirection moveDirection;
    private double hp;

    private int targetId;
    @JsonProperty("actionType")
    @JsonFormat(shape = JsonFormat.Shape.NUMBER_INT)
    private ActionType actionType;

    private double initHp;
    private int level;

    @JsonProperty("isConnected")
    private boolean isConnected;
    @JsonProperty("classId")
    @JsonFormat(shape = JsonFormat.Shape.NUMBER_INT)
    private CharacterClass classId;
}
